helixbass/tree-sitter-grep
# Errors for no files searched, invalid paths

In this PR:
- in working on the `ignore` refactor I ran into stuff where it seemed like cleaning up how we're handling/tracking the state of the combination of the `--language` flag, specified path arguments (files and/or directories) and whether we could parse the query would be a good "first step"

To test:
Per additional test cases, when it failed to parse the query for the specified or any of the auto-detected languages (and "nothing else went wrong") you should now see which languages it tried to parse the query in in the error message
When it didn't search any files (because there were no recognized file types in the current directory or specified paths) it should show an error message that it didn't search any files
When you specify a nonexistent directory and/or file path on the command line you should see corresponding file not found error messages
